pkg/search/postgres/common.go

```diff
@@ -882,6 +908,10 @@ func RunDeleteRequestForSchema(ctx context.Context, schema *walker.Schema, q *v1
 
 // RunDeleteRequestReturningIDsForSchema executes a request for the delete query against the database returning IDs.
 func RunDeleteRequestReturningIDsForSchema(ctx context.Context, schema *walker.Schema, q *v1.Query, db postgres.DB) ([]string, error) {
+	if q == nil {
+		q = searchPkg.EmptyQuery()
```

Comment:
hmm, is this dangerous? What happens if someone passed nil, do we delete everything or does the line below catch it ? If the line below catches it can, we just encode this into the logic to `return nil, nil`

Reply:
Good catch. I second that delete should not translate `nil` to delete all, that should be the behavior of `RunDeleteRequestReturningIDsForSchema ` and `RunDeleteRequestForSchema`. Honestly, I am not sure how I missed `Delete` from this function name and not the function above. 

Nonetheless, while verifying the behavior I found that the DELETE relies on an invariant that the caller must send a non-nil query. This invariant seems to be weak and requires mandatory validations earlier in the call stack. In other words, all nil queries are treated as select all queries. I will open a PR to address that.